core/templates/dev/head/pages/exploration_player/ProgressNavDirective.js

```diff
@@ -23,20 +23,41 @@ oppia.directive('progressNav', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/pages/exploration_player/' +
         'progress_nav_directive.html'),
+      scope: {
+        submitFn: '&',
+        onClickContinueButton: '&'
+      },
       controller: [
         '$scope', '$rootScope', 'PlayerPositionService',
-        'PlayerTranscriptService',
+        'PlayerTranscriptService', 'ExplorationPlayerService',
+        'ExplorationPlayerStateService', 'CONTINUE_BUTTON_FOCUS_LABEL',
         function($scope, $rootScope, PlayerPositionService,
-          PlayerTranscriptService) {
-          $scope.currentCardIndex = PlayerPositionService.getActiveCardIndex();
-          $scope.transcriptLength = PlayerTranscriptService.getNumCards();
+          PlayerTranscriptService, ExplorationPlayerService,
+          ExplorationPlayerStateService, CONTINUE_BUTTON_FOCUS_LABEL) {
+          $scope.CONTINUE_BUTTON_FOCUS_LABEL = CONTINUE_BUTTON_FOCUS_LABEL;
+
+          var update = function() {
+            var index = PlayerPositionService.getActiveCardIndex();
+            $scope.activeCardIndex = index;
+            $scope.activeCard = PlayerTranscriptService.getCard(index);
+            $scope.transcriptLength = PlayerTranscriptService.getNumCards();
```

Comment:
Throughout this method, don't declare stuff on $scope unless you really need to. E.g. I don't think you need to do that for transcriptLength.

Declaring stuff on $scope exposes it to the directive HTML and it would be better not to expose properties needlessly.

Reply:
Fixed.